디벨롭 커밋 푸시

Park-Sang-soo opened the pull request.

Commit:
디벨롭 커밋 푸시

Commit:
두번째 수정테스트 주석추가

Commit:
수정테스트

LimKyunTae commented on blackyakb2c/src/main/java/com/blackyak/b2c/vo/ProductVo.java:
```diff
@@ -4,7 +4,8 @@
 
 import lombok.Getter;
 import lombok.Setter;
-
```
@JsonProperty(value="compCd")
로 썼던데 API 필드명이 그럼 compCd로 제공됨?

Park-Sang-soo replied:
맞음!

{
  "compCd": "003",
  "refCompCd": "003",
  "goodsGb": "G",
  "brndCd": "BY",
  "prodCd": "ABYSHX1908",
  "prodNm": "뉴 드리븐 Ⅱ GTX",
  "prodDiv": "A",

LIM-KT replied:
확인 완료

LimKyunTae commented:
현재 프로젝트의 폴더 구조가 너무 이상함
주제영역별로 폴더구조를 재생성 할것

ProductDao는 ProductMapper로 변경


b2c
- product
    - controller
    - service
    - mapper
- member
- common
    - util
    - code
    - exception
    - config

Commit:
트리구조변경 및 vo와 dto로 변경

Commit:
다시 푸시

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/service/impl/ProductServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+import com.blackyak.b2c.product.vo.ProductVo;
+
+@Service
+public class ProductServiceImpl implements ProductService {
```
요즘은 Impl 안만듬 삭제할것

Park-Sang-soo replied:
삭제완료

LIM-KT replied:
확인 완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/controller/ProductController.java:
```diff
@@ -1,37 +1,46 @@
+	*/
 	
-	
+	@GetMapping("/product-info")
```
스웨거 사용하는 것으로 설정 잡아놓고 하나도 스웨거 관련 어노테이션이 없음
controller / VO에 스웨거 어노테이션으로 이쁘게 정리해보셈

Park-Sang-soo replied:
vo, dto 제대로 못받아오고 있어서, 해결부터 하겠음...

LIM-KT replied:
당연히 vo 못받아오지 컨트롤러에 코멘트 해뒀음

Park-Sang-soo replied:
기존에 적용했던 스웨거2에서 @Schema 어노테이션을 사용할 수 없어서 스웨거3로 변경적용함

LIM-KT replied:
항상 버전은 높은 버전으로 고려하셈

LIM-KT replied:
확인 완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/mapper/dto/ProductDto.java:
```diff
@@ -0,0 +1,14 @@
+@Setter
+public class ProductDto {
+	
+	public String comp_cd;	
```
DB 컬럼이 스네이크 케이스이면 자동으로 카멜케이스로 바꿀수있는 mybatis 설정이 있음
적용 해보셈

ex. product 테이블에 comp_cd 라는 컬럼이 있을때
public String compCd; 로 선언해서 받을 수 있음

Park-Sang-soo replied:
찾아보니, config 파일부터 생성하여 설정 후에 해야하는 것 같던데
config파일의 경로는 현재 트리구조 안에 common→config 안에 생성하여 설정하면 되는지????

LIM-KT replied:
ㅇㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/vo/ProductVo.java:
```diff
@@ -0,0 +1,105 @@
+	
+	@Getter
+	public static class Request {
+		private String comp_cd;
```
compCd와 같이 카멜케이스로 바꿀것

LIM-KT replied:
화면에서 comp_cd로 넘어온다면 compCd로 받게끔 할수가 있음
해보셈

LIM-KT replied:
근데 웃긴게 응답에보면 응답 필드는 compCd 로 넘겨주니까 화면에서는 compCd로 쓸텐데
요청때는 comp_cd로 요청을 함? 통일을 할 필요가 있음

Park-Sang-soo replied:
완료

LIM-KT commented:
application.properties -> application.yml로 바꿀것

LIM-KT commented:
myBatis 설정을 설정파일에서 하지말고
별도로 bean 등록해서 할것

`@Configuration
@MapperScan("com.xxx.xxx....)
public class MyBatisConfig {....}`

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/controller/ProductController.java:
```diff
@@ -1,37 +1,46 @@
 	
-	
+	@GetMapping("/product-info")
+	public ProductDto productInfo(@RequestBody ProductVo Request) throws Exception {
```
DTO는 service, mapper layer에서만 사용할것
아래처럼 바꿀것
	public ProductVo.Response productInfo(@RequestBody ProductVo.Request request) throws Exception {

LIM-KT commented:
- application.properties -> application.yml로 바꿀것

- myBatis 설정을 설정파일에서 하지말고 별도로 bean 등록해서 할것
 >>> @Configuration @MapperScan("com.xxx.xxx....) public class MyBatisConfig {....}

Commit:
의존성 주입의 문제로써, @Autowired 어노테이션을 사용해서 해결하려 하였으나,
해결되지 않아
private static ProductMapper productMapper;

    public ProductService(ProductMapper productMapper) {
        this.productMapper = productMapper;
    }
생성자로 주입하니 해결됨.

Park-Sang-soo commented:
Request URL
http://localhost:8080/products/product/ABYSHX3915?colrCd=BE&compCd=003&sizeCd=235

Response body
{
  "compCd": "003",
  "refCompCd": "003",
  "goodsGb": "G",
  "brndCd": null,
  "prodCd": "ABYSHX3915",
  "prodNm": "343 아크 GTX",
  "prodDiv": "A",
  "prodDivNm": "신발",
  "goodKindCd": "SH",
  "goodKindNm": "신발",
  "planYear": "S",
  "planYearNm": "2023",
  "sesnCd": "X",
  "sesnNm": "비시즌",
  "sexDiv": "U",
  "sexDivNm": "공용",
  "prodLineCd": "3",
  "prodLineNm": "ACTIVE TRAIL(신발)",
  "colrCd": "BE",
  "colrNm": "BEIGE",
  "sizeCd": "235",
  "normRtalPrce": 259000,
  "madeYymm": null,
  "origNatCd": "KR",
  "origNatNm": "국내",
  "dispOrd": 0
}

Commit:
swagger2에서 swagger3로 변경되면서 어노테이션이 많이 변경됨
@Schema는 swagger3에서 사용가능한 어노테이션

Commit:
application.properties -> application.yml 변경완료
properties은 비계층적관리 key->value 구조
yml은 계층적관리

Commit:
테스트 오류라서 재 푸시

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/common/config/SwaggerConfig.java:
```diff
@@ -29,7 +55,7 @@ public Docket SwaggerApi(ServletContext servletContext) {
                     .groupName("Test")
                     .select()
-                    .apis(RequestHandlerSelectors.basePackage("com.blackyak.b2c.web"))
+                    .apis(RequestHandlerSelectors.basePackage("com.blackyak.b2c.product"))
```
해당 프로젝트가 product API 만 존재함????
order / event .... 등등 다른 API가 만들어질 가능성이 없음????

com.balckyak.b2c
 - common
 - product
이렇게 구조가 되어있는데

com.balckyak.b2c
 - common
 - api
 - - prodcut
 - - order
 - - event
 이런식으로 api 패키지를 추가하는게 낫지 않음?
 현재대로라면 product 폴더만 스웨거 api 지원이 되잖아

Park-Sang-soo replied:
구조 변경 완료

LIM-KT replied:
굳

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/controller/ProductController.java:
```diff
@@ -0,0 +1,54 @@
+	
+	@Tag(name = "Product")
+	@Operation(summary = "제품상세정보조회", description = "제품코드를 받아 제품상세정보를 조회하는 API")
+	@ApiResponses({
```
지우셈 ExceptionHandler 구현도 안해놨는데 뭔 벌써 에러 리턴 정의함?
나중에 ExceptionHandler 구현 할꺼니 그때 생각해보면됨
ExceptionHandler 구현 하더라도 apiResponses에 덕지 덕지 저렇게 붙여놓진 않음

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/controller/ProductController.java:
```diff
@@ -0,0 +1,54 @@
+        @ApiResponse(responseCode = "500", description = "INTERNAL SERVER ERROR")
+        
+	})	
+	@GetMapping("/product/{prodCd}") //@Parameter(description = "제품코드", required = true) 
```
불필요한 주석은 남기지 말것
주석없이 코딩해서하는게 제일 좋은 방법임(가독성있는 변수명, 코드 등을 해야한다는 소리임)
주석은 로직이 복잡하거나 아니면 어떤 특정 요인에 의해 어쩔수없이 변수명이 다르다거나 의미하는 바가 명확하지 않을떄에 달것

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/mapper/ProductMapper.java:
```diff
@@ -0,0 +1,14 @@
+@Mapper
+public interface ProductMapper {
+	
+	//List<ProductVo> selectProductList(String prod_cd);		
```
주석삭제

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/mapper/dto/ProductDto.java:
```diff
@@ -0,0 +1,42 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductDto {
+	
```
myBatis 카멜케이스 빨리 적용할것 보기 싫음

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/service/ProductService.java:
```diff
@@ -0,0 +1,68 @@
+	//List<ProductVo> selectProductList(String prod_cd) throws Exception;
+		
+	//ProductDto selectProductInfo(ProductVo request);
+	
```
불필요 주석 삭제

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/service/ProductService.java:
```diff
@@ -0,0 +1,68 @@
+		System.out.println("Service >>>"+request.getProdCd());
+		System.out.println("Service >>>"+request.getColrCd());
+		System.out.println("Service >>>"+request.getSizeCd());
+		
```
테스트시에만 sysout 찍을것 삭제요망
추후 log 모듈 붙일것

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/vo/ProductVo.java:
```diff
@@ -0,0 +1,150 @@
+    @NoArgsConstructor
+    @AllArgsConstructor
+	public static class Response {
+				
```
comp_cd 변수명 뭐임?
companyCode로 바꿀것

db에서 뺴오는 dto는 db컬럼 자체가 자연어가 아니니 어쩔수 없고
서비스로직이나 api 응답 로직은 모두 자연어로 바꿀것

LIM-KT replied:
dto = compCd
vo = companyCode / jsonPropertiy는 클라이언트에서 필요한 명칭으로(compCd)

무조건 바꿀것

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/product/vo/ProductVo.java:
```diff
@@ -0,0 +1,150 @@
+	public static class Request {
+		
+		@ApiModelProperty(value = "회사코드")
+		private String compCd;		
```
변수명 바꿀것
companyCode
아래 변수도 모두다

@JsonProperty 쓰면됨 응답과 똑같이

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/resources/application.yml:
```diff
@@ -0,0 +1,18 @@
+
+      
+mybatis:
+#Xml
```
빨리 마이바티스 설정 java config로 바꿀 것

Park-Sang-soo replied:
java config 파일은 common아래의 config에 파일 생성하면 되는지?????

LIM-KT replied:
ㅇㅇ

LIM-KT commented on blackyakb2c/src/main/resources/mapper/ProductMapper.xml:
```diff
@@ -3,8 +3,9 @@
-  <select id="selectProductList" resultType="com.blackyak.b2c.vo.ProductVo">
+<mapper namespace="com.blackyak.b2c.product.mapper.ProductMapper">
+	
+  <select id="selectProductList" resultType="com.blackyak.b2c.product.mapper.dto.ProductDto">
```
productMapper에도 없는데 왜 있는거임
삭제

Park-Sang-soo replied:
완료

LIM-KT replied:
ㅇ

LIM-KT commented on blackyakb2c/src/main/resources/mapper/ProductMapper.xml:
```diff
@@ -13,11 +14,12 @@
   </select>
-  <select id="selectProductInfo" parameterType="com.blackyak.b2c.vo.ProductVo" resultType="com.blackyak.b2c.vo.ProductVo">
+  
+  <select id="selectProductInfo" parameterType="com.blackyak.b2c.product.mapper.dto.ProductDto" resultType="com.blackyak.b2c.product.mapper.dto.ProductDto">
```
parameterType resultType
com.blackyak.b2c.product.mapper.dto.ProductDto이렇게 패키지 다쓰지말고

ProductDto 이렇게만써도됨 바꿔서 안되면 찾아보셈 별거아님

Park-Sang-soo replied:
완료(정상작동 확인)

LIM-KT replied:
ㅇㅇ

LIM-KT commented on blackyakb2c/src/main/resources/application.yml:
```diff
@@ -0,0 +1,18 @@
+#Xml
+  mapper-locations: classpath:mapper/*.xml
+#Dto
+  type-aliases-package: com.blackyak.b2c.product.mapper.dto
```
이렇게 세팅하면 추후에 order 가 생겼을때 어떻게 할꺼임?
com.blackyak.b2c.order.mapper.dto 멀티로 세팅할꺼임?

이거 com.blackyak.b2c 패키지아래에 있는 common 폴더에 db 패키지 만들어서 통합 관리하면 좋을듯

com.blackyak.b2c.common.db
 - mapper
 - - product
 - - - ProductMapper.java
 - - order....
 - dto
 - - proudct
 - - - ProductDto
 - - order.....

그리고 위의 mapper , dto 패키지 안에 product, order 등등을 굳이 나눠야 하나 잘 생각해보셈
product 라는 관심사에 mapper나 dto가 여러개가 생길지? 이건 프로젝트별로 다름(내생각엔 안나뉠거 같음)
많이 안생긴다면

com.blackyak.b2c.common.db
 - mapper
 - - ProductMapper.java
 - - OrderMapper.java
 - dto
 - - ProductDto.java
 - - OrderDto.java

이런식이 될듯

Park-Sang-soo replied:
현재 product 패키지 안에 controller, mapper, service, vo를 하위 패키지로 생성해놓은 상태인데
새로생기는 order나 member등은 product의 구조처럼 생성하면 안되는 것인지???
이해를 못했음.....

LIM-KT replied:
나중에 order 라는 패키지를 생성했을때
매퍼를 어떻게 정의할꺼임?

현재
`  type-aliases-package: com.blackyak.b2c.product.mapper.dto` 이렇게 세팅해놔서 product 패키지 안에있는 매퍼만 세팅이 가능한데 나중에 order 패키지가 생겼을 때 order 하위에 mapper가 있으면 중복 세팅이 불가하잖아

그러니 각 패키지에 있는 mapper를 common/db 패키지에 넣어서 관리 하는게 어떻겠냐는 거지

Park-Sang-soo replied:
완료(정상 작동확인)

Commit:
23.04.14 패키지트리 구조변경 및 필요없는 주석제거

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/api/product/controller/ProductController.java:
```diff
@@ -0,0 +1,34 @@
+	@Tag(name = "Product")
+	@Operation(summary = "제품상세정보조회", description = "제품코드를 받아 제품상세정보를 조회하는 API")
+	@GetMapping("/product/{prodCd}")
+	public ProductVo.Response getProduct(@Parameter(description = "제품코드") @PathVariable("prodCd") String prodCd, ProductVo.Request request){		
```
항상 가독성이 좋게 코딩하는 습관을 기르셈

```java
public ProductVo.Response getProduct(
    @Parameter(description = "제품코드")
    @PathVariable("prodCd") String prodCd,
    ProductVo.Request request){
```

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/api/product/mapper/dto/ProductDto.java:
```diff
@@ -0,0 +1,42 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductDto {
+	
```
`private String comp_cd;`
->
`private String compCd;`

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/api/product/service/ProductService.java:
```diff
@@ -0,0 +1,59 @@
+        this.productMapper = productMapper;
+    }
+	
+	public static ProductVo.Response selectProductInfo(ProductVo.Request request) {
```
서비스 로직 선언 위에 항상 주석을 달것 이미지 처럼

![image](https://user-images.githubusercontent.com/130113000/232080467-ef134085-96c6-40c4-9615-c3358191b8a8.png)

Park-Sang-soo replied:
완료 명심하겠음! 습관들이겠음

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/api/product/vo/ProductVo.java:
```diff
@@ -0,0 +1,150 @@
+		private String compCd;		
+		
+		@ApiModelProperty(value = "제품코드", hidden = true)
+		private String prodCd;		
```
hidden = true는 잘했음 그러나 빨리좀 바꾸셈 결벽증 때문에 코드 보기가 싫음

```java
@ApiModelProperty(value = "제품코드", hidden = true)
@JsonProperty(value="prodCd")
private String productCode;	
```

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/api/product/vo/ProductVo.java:
```diff
@@ -0,0 +1,150 @@
+				
+		@JsonProperty(value="compCd")
+		@Schema(description = "회사코드", example = "003", required = true)
+		private String comp_cd;
```
private String comp_cd -> companyCode

Park-Sang-soo replied:
완료

LIM-KT commented on blackyakb2c/src/main/java/com/blackyak/b2c/common/config/SwaggerConfig.java:
```diff
@@ -1,24 +1,50 @@
+                .build();
+    }
+	
+	/*
```
주석은 삭제할것
필요해서 남겨놨다면 아래와같이 사유를 자세히 적어야함

// TODO : xxx 로 인한 주석처리

LIM-KT commented on blackyakb2c/src/main/resources/mapper/ProductMapper.xml:
```diff
@@ -3,21 +3,13 @@
-  <select id="selectProductInfo" parameterType="com.blackyak.b2c.vo.ProductVo" resultType="com.blackyak.b2c.vo.ProductVo">
+<mapper namespace="com.blackyak.b2c.api.product.mapper.ProductMapper">
+  
+  <select id="selectProductInfo" parameterType="com.blackyak.b2c.api.product.mapper.dto.ProductDto" resultType="com.blackyak.b2c.api.product.mapper.dto.ProductDto">
```
parameterType resultType alias로 짧게 쓰는 방법 찾아서 적용할것

Commit:
VO 변수명 풀네임으로 변경

Commit:
dto 변수명 변경

Commit:
ProductMapper.xml의 parameterType, resultType 경로 간소화

Commit:
MybatisConfig.java로 설정파일 변경

Park-Sang-soo commented:
MybatisConfig.java에서 궁금한게 있음.
setTypeAliasesPackage에서 resultType의 경로를 입력하는데,
현재는 Product하나라 
sessionFactoryBean.setTypeAliasesPackage("com.blackyak.b2c.api.product.mapper.dto");
으로 경로를 지정했는데, 만약에 member나 order가 더 생기면 그때마다 경로를 추가로 지정해줘야하는지???

Commit:
mapper, dto를 common.db로 이동 및 서비스 로직 선언 위에 주석 작성함

Park-Sang-soo commented:
처음에 말한 리스트

시큐리티 적용
인터셉터 만들기
필터만들기
aop 적용
맵스트럭트 적용해서 vo 매핑
에러핸들러
공통에러처리

중에 젤 상위부터 적용하면 될지?

Commit:
불필요한 주석제거

Park-Sang-soo commented:
지금 만드는 api는 자사몰↔erp 데이터를 조회하는 걸 만드는데, 
시큐리티는 로그인이나 이런 개인정보 접근이나 암호화 때문에 사용하는게 아닌지?
이전에 타 회사에서 제공받은 api로 정보를 가져오게 만들때는 그쪽에서 무슨 인증키 같은걸 같이 줬는데,
그걸 만들어서 접근하게끔 할 수 있는 방법은 어떻게 해야하는지?
